src/builtin.jq

```diff
@@ -139,13 +139,18 @@ def gsub($re; s): sub($re; s; "g");
 ########################################################################
 # generic iterator/generator
 def while(cond; update):
-     def _while:
-         if cond then ., (update | _while) else empty end;
-     _while;
+  def _while:
```

Comment:
We need to define a code style for jq.  I've become very happy with 4-space indentation except for 2 spaces for `|`, like this:

```jq
    .
  | stuff
```

I think until we decide on a style for jq code, maybe it'd be best to not restyle existing jq code.

Reply:
I was just trying to ensure while/whilst/until were consistently formatted, admittedly in the 2-space style -- two spaces saves space-time, no?  I thought that was important for builtin.jq ...

Reply:
Changing style of surrounding code to be more consistent is something I like to do -though it can be irritating because it clutters diffs and complicates use of `git blame`-, but in this case I'm not sure what style we ought to adopt, and my own jq-coding style has change over time, so for `src/builtin.jq` I would prefer we don't gratuitously re-style until we decide on a style.  Conversely, I'm not asking you to adopt a style for jq code for now either -- if eventually we have to restyle all of `src/builtin.jq`, so be it.

(For C code in jq the style is fairly consistent already, and very much in @stedolan's C style.)

Reply:
@nicowilliams - In case you haven't noticed, @itchyny's builtin.jq in gojq uses the two-space style :-)

https://github.com/itchyny/gojq/blob/main/builtin.jq

Reply:
@pkoppstein no, I've not noticed.  It's on my list to eventually go take a look at `fq`, `jqjq`, `gojq`, `rq`, and others, but my time is limited.  What they do for style is interesting, but what they do for semantics is much more interesting as we want jq-alikes to interoperate.  As for style, like I said, my own jq style has evolved and I think we should discuss that separately and not gratuitously restyle `src/builtin.jq` _for the time being_ even though I otherwise would prefer that we do restyle as we go.